subsys/net/ip/tcp2.c

```diff
@@ -0,0 +1,1584 @@
+/*
+ * Copyright (c) 2018-2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(net_tcp, CONFIG_NET_TCP_LOG_LEVEL);
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <zephyr.h>
+#include <net/net_pkt.h>
+#include <net/net_ip.h>
+#include <net/net_context.h>
+#include "connection.h"
+#include "net_stats.h"
+#include "tcp2_priv.h"
+
+static int tcp_rto = 500; /* Retransmission timeout, msec */
```

Comment:
In some future PR, the hard coded values need to be changed to something that can be tweaked from Kconfig.